src/common/options/osd.yaml.in

```diff
@@ -133,6 +133,54 @@ options:
   - osd_recovery_sleep
   flags:
   - runtime
+- name: osd_recovery_sleep_degraded
+  type: float
+  level: advanced
+  desc: Time in seconds to sleep before next recovery or backfill op when pg is degraded.
+    This setting overrides _ssd, _hdd, and _hybrid if non-zero.
+  fmt_desc: Time in seconds to sleep before the next recovery or backfill op when pg
+    is degraded. Increasing this value will slow down recovery operation while client
+    operations will be less impacted.
+  default: 0
+  flags:
+  - runtime
+  with_legacy: true
```

Comment:
I'm not sure it makes much sense to replicate existing mess with legacy/non-legacy parameters for osd_recover_sleep. IMO any new parameter can be defined as non-legacy one.